cmd/bucket-lifecycle.go

```diff
@@ -360,6 +383,7 @@ const (
 // 1. when a restored (via PostRestoreObject API) object expires.
 // 2. when a transitioned object expires (based on an ILM rule).
 func expireTransitionedObject(ctx context.Context, objectAPI ObjectLayer, oi *ObjectInfo, lcOpts lifecycle.ObjectOpts, action expireAction) error {
+	traceFn := globalLifecycleSys.trace(*oi)
```

Comment:
we should be doing `defer traceFn` such that we can capture the other returns as well.

Reply:
deferring auditLogLifecycle call since traceFn is passed into this.